### PR TITLE
CIRC-2006 - Add tenant's timezone in the payload that will be used to create circ log entries.

### PR DESCRIPTION
**Pull request by gurleenkaurbp**
## Purpose
In order to fix MODAUD-176 tenant's timezone value is required to convert the return date and system return date and then compare it to created the backdate description in circ log.

## Approach
Added timezone in the payload before being sent to mod-audit for creating circ log entries.

#### TODOS and Open Questions
- [ ] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/CheckInContext.java [hunk 1]
```diff
@@ -240,4 +267,8 @@ public ZonedDateTime getCheckInProcessedDateTime() {
   public ItemStatus getItemStatusBeforeCheckIn() {
     return itemStatusBeforeCheckIn;
   }
+
+  public ZoneId getTimeZone() {
+    return timeZone;
+  }
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/CheckInContext.java:273]: Same here, can use `@Getter` annotation instead and remove other trivial getters. Again, you can check `LoanAndRelatedRecords` as one of the examples.

File: src/main/java/org/folio/circulation/domain/CheckInContext.java [hunk 2]
```diff
@@ -182,7 +192,24 @@ public CheckInContext withItemStatusBeforeCheckIn(ItemStatus itemStatus) {
       this.loggedInUserId,
       this.checkInProcessedDateTime,
       this.inHouseUse,
-      itemStatus);
+      itemStatus,
+      this.timeZone);
+  }
+
+  public CheckInContext withTimeZone(ZoneId timeZone) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/CheckInContext.java:199]: Can we use `@With` class annotation and also remove other `with...()` methods from this class? We usually do this whenever we need to add one more `with...()` method. Example - `LoanAndRelatedRecords` class

Comment (gurleenkaurbp) [reply src/main/java/org/folio/circulation/domain/CheckInContext.java]: Done

File: src/main/java/org/folio/circulation/domain/CheckInContext.java (hunk 1 above)
Comment (gurleenkaurbp) [reply src/main/java/org/folio/circulation/domain/CheckInContext.java]: Done

**Review by alexanderkurash — APPROVED** [review]

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Review by SinghAdes — APPROVED** [review]

**Pull request merged**